test/spanner.ts

```diff
@@ -1314,6 +1314,43 @@ describe('Spanner with mock server', () => {
           );
         });
 
+        it('should retry UNAVAILABLE during streaming with txn ID from inline begin response with parallel queries', async () => {
+          const err = {
+            message: 'Temporary unavailable',
+            code: grpc.status.UNAVAILABLE,
+            streamIndex: index,
+          } as MockError;
+          spannerMock.setExecutionTime(
+            spannerMock.executeStreamingSql,
+            SimulatedExecutionTime.ofError(err)
+          );
+          const database = newTestDatabase();
+
+          await database.runTransactionAsync(async tx => {
+            await Promise.all([tx!.run(selectSql), tx!.run(selectSql)]);
+            await tx.commit();
+          });
+          await database.close();
+
+          const requests = spannerMock
+            .getRequests()
+            .filter(val => (val as v1.ExecuteSqlRequest).sql)
+            .map(req => req as v1.ExecuteSqlRequest);
+          assert.strictEqual(requests.length, 3);
+          assert.ok(
+            requests[0].transaction?.begin!.readWrite,
+            'inline txn is not set.'
+          );
+          assert.ok(
+            requests[1].transaction!.id,
+            'Transaction ID is not used for retries.'
+          );
+          assert.ok(
+            requests[1].resumeToken,
+            'Resume token is not set for the retried'
+          );
+        });
+
```

Comment:
Can we add a test that verifies the behavior if the initial `ExecuteStreamingSql` call succeeds, but the stream then fails with `UNAVAILABLE` after receiving the first result? That should lead to the following behavior:
1. `ExecuteStreamingSql` is retried, but __without__ a `transaction.begin` and __with__ a `transaction.id` (and with a resume token)
2. There's also no extra `BeginTransaction` RPC

Reply:
(It could be that we already have such a test, so check for that first)

Reply:
Ack